### PR TITLE
gitignore: add more rules and make others more generic

### PR DESCRIPTION
**Pull request by sebix**
adding more temporary files of editors
change some existing rules to make them more generic, matching all
directory levels
group the entries by type, document that

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,29 +1,30 @@
...
+venv/
+__pycache__/
+*.pyc
+node_modules/
+dist/
+build/
 
+# temporary files of editors
 .DS_Store
-.*.swp
```
Comment (milankowww) [inline .gitignore:10]: The proper pattern would be `.*.sw?`, not `*.sw?`

File: .gitignore [hunk 2]
```diff
@@ -1,29 +1,30 @@
...
-.idea
-.vscode
+# settings of editors
+.idea/
+.vscode/
 *.suo
 *.ntvs*
 *.njsproj
 *.sln
-*.sw?
```
Comment (milankowww) [inline .gitignore:29]: see the comment above

**Review by milankowww — CHANGES_REQUESTED** [review]
a minor change to .gitignore patterns would be appreciated

File: .gitignore (hunk 2 above)
Comment (sebix) [reply .gitignore:29]: I didn't change that line - the newline was missing here

File: .gitignore (hunk 1 above)
Comment (sebix) [reply .gitignore:10]: fixed

**Review by dodancs — APPROVED** [review]
Okay, the changes seem to be right. Thanks!

**Pull request merged**